addons/rating/models/rating_mixin.py

```diff
@@ -42,12 +47,38 @@ def _compute_rating_percentage_satisfaction(self):
                 grades_per_parent[parent_id]['okay'] += item['__count']
             else:
                 grades_per_parent[parent_id]['bad'] += item['__count']
+            rating_scores_per_parent[parent_id] += rating * item['__count']
 
         # compute percentage per parent
         for record in self:
             repartition = grades_per_parent.get(record.id, default_grades)
-            record.rating_count = sum(repartition.values())
-            record.rating_percentage_satisfaction = repartition['great'] * 100 / sum(repartition.values()) if sum(repartition.values()) else -1
+            rating_count = sum(repartition.values())
+            record.rating_count = rating_count
+            record.rating_percentage_satisfaction = repartition['great'] * 100 / rating_count if rating_count else -1
+            record.rating_avg = rating_scores_per_parent[record.id] / rating_count if rating_count else 0
+            record.rating_avg_percentage = record.rating_avg / 5
+
+    def _search_rating_avg(self, operator, value):
+        if operator not in ['=', '!=', '<=', '<', '>', '>=']:
+            raise NotImplementedError('This operator %s is not supported in this search method.' % operator)
+        if not isinstance(value, (int, float)):
+            raise NotImplementedError('The value must be an integer or a float value.')
+        where_write_date_query = ''
+        if self._rating_satisfaction_days and isinstance(self._rating_satisfaction_days, int):
```

Comment:
Why this isinstance ? Isn't it a bit too defensive for something that we basically control as it is defined at class level ?

Reply:
Indeed, you are right. 😇